zenoh/src/net/routing/interceptor/access_control.rs

```diff
@@ -0,0 +1,282 @@
+            .or_else(|| ctx.full_expr())?;
+        if let NetworkBody::Push(Push {
+            payload: PushBody::Put(_),
+            ..
+        })
+        | NetworkBody::Request(Request {
+            payload: RequestBody::Put(_),
+            ..
+        })
+        | NetworkBody::Response(Response {
+            payload: ResponseBody::Put(_),
+            ..
+        }) = &ctx.msg.body
+        {
+            let mut decision = self.default_decision;
+
+            for subject in &self.interface_list {
+                match self.policy_enforcer.policy_decision_point(
+                    *subject,
+                    Action::Put,
+                    key_expr,
+                    self.default_decision,
+                ) {
+                    Ok(true) => {
+                        decision = true;
+                        break;
+                    }
+                    Ok(false) => continue,
+                    Err(e) => {
+                        log::error!("Authorization incomplete due to error {}", e);
+                        return None;
+                    }
+                }
+            }
+
+            if !decision {
+                log::warn!("{} is unauthorized to Put", self.zid);
+                return None;
+            }
+            log::info!("{} is authorized access to Put", self.zid);
+        } else if let NetworkBody::Request(Request {
+            payload: RequestBody::Query(_),
+            ..
+        }) = &ctx.msg.body
+        {
+            let mut decision = self.default_decision;
+            for subject in &self.interface_list {
+                match self.policy_enforcer.policy_decision_point(
+                    *subject,
+                    Action::Get,
+                    key_expr,
+                    self.default_decision,
+                ) {
+                    Ok(true) => {
+                        decision = true;
+                        break;
+                    }
+                    Ok(false) => continue,
+                    Err(e) => {
+                        log::error!("Authorization incomplete due to error {}", e);
+                        return None;
+                    }
+                }
+            }
+            if !decision {
+                log::warn!("{} is unauthorized to Query/Get", self.zid);
+                return None;
+            }
+
+            log::info!("{} is authorized access to Query", self.zid);
+        }
+        Some(ctx)
+    }
+}
+
+impl InterceptorTrait for EgressAclEnforcer {
+    fn compute_keyexpr_cache(&self, key_expr: &KeyExpr<'_>) -> Option<Box<dyn Any + Send + Sync>> {
+        Some(Box::new(key_expr.to_string()))
+        // None
+    }
+    fn intercept(
+        &self,
+        ctx: RoutingContext<NetworkMessage>,
+        cache: Option<&Box<dyn Any + Send + Sync>>,
+    ) -> Option<RoutingContext<NetworkMessage>> {
+        let key_expr = cache
+            .and_then(|i| i.downcast_ref::<String>().map(|e| e.as_str()))
```

Comment:
Similar comment as before, error of `downcast_ref` is silently converted into as string.